src/main/webapp/app/shared/metis/posting-header/answer-post-header/answer-post-header.component.html

```diff
@@ -37,48 +38,49 @@
-                        <fa-icon
-                            class="resolved"
-                            [icon]="faCheck"
-                            [ngbTooltip]="
-                                isAtLeastTutorInCourse || isAuthorOfOriginalPost
-                                    ? ('artemisApp.metis.unmarkAsResolvingAnswerTooltip' | artemisTranslate)
-                                    : ('artemisApp.metis.resolvingAnswerTooltip' | artemisTranslate)
-                            "
-                        />
-                    </div>
-                } @else {
-                    @if (isAtLeastTutorInCourse || isAuthorOfOriginalPost) {
+    @if (!isDeleted()) {
+        <div class="d-flex gap-2">
+            @if (mayEditOrDelete) {
+                <fa-icon
+                    size="xs"
+                    class="editIcon clickable icon"
+                    [ngbTooltip]="'artemisApp.metis.editPosting' | artemisTranslate"
+                    [icon]="faPencilAlt"
+                    (click)="openPostingCreateEditModal.emit()"
+                />
+            }
+            @if (mayEditOrDelete) {
+                <jhi-confirm-icon
+                    iconSize="xs"
+                    (confirmEvent)="deletePosting()"
+                    [ngClass]="'deleteIcon clickable icon'"
+                    [initialTooltip]="'artemisApp.metis.deleteAnswer' | artemisTranslate"
+                    [confirmTooltip]="'artemisApp.metis.confirmDeleteAnswer' | artemisTranslate"
+                />
+            }
+            @if (!isAnswerOfAnnouncement) {
+                <div id="toggleElement" class="resolve" [ngClass]="isAtLeastTutorInCourse || isAuthorOfOriginalPost ? 'clickable' : ''" (click)="toggleResolvesPost()">
+                    @if (posting.resolvesPost) {
                         <div>
-                            <fa-icon class="icon notResolved" [icon]="faCheck" [ngbTooltip]="'artemisApp.metis.markAsResolvingAnswerTooltip' | artemisTranslate" />
+                            <fa-icon
+                                class="resolved"
+                                [icon]="faCheck"
+                                [ngbTooltip]="
+                                    isAtLeastTutorInCourse || isAuthorOfOriginalPost
+                                        ? ('artemisApp.metis.unmarkAsResolvingAnswerTooltip' | artemisTranslate)
+                                        : ('artemisApp.metis.resolvingAnswerTooltip' | artemisTranslate)
+                                "
+                            />
                         </div>
+                    } @else {
+                        @if (isAtLeastTutorInCourse || isAuthorOfOriginalPost) {
+                            <div>
+                                <fa-icon class="icon notResolved" [icon]="faCheck" [ngbTooltip]="'artemisApp.metis.markAsResolvingAnswerTooltip' | artemisTranslate" />
+                            </div>
+                        }
                     }
```

Comment:
```suggestion
                    } @else if (isAtLeastTutorInCourse || isAuthorOfOriginalPost) {
                        <div>
                            <fa-icon class="icon notResolved" [icon]="faCheck" [ngbTooltip]="'artemisApp.metis.markAsResolvingAnswerTooltip' | artemisTranslate" />
                        </div>
                    }
```

Reply:
Thanks for the feedback! Moved the if into an else if block in this [commit](https://github.com/ls1intum/Artemis/pull/9624/commits/82a35634c5635a96949b32342c0bee087f8da25a).